src/main/java/org/terasology/logic/health/DamageResistComponent.java

```diff
@@ -0,0 +1,39 @@
+package org.terasology.logic.health;
+
+import org.terasology.entitySystem.Component;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+public class DamageResistComponent implements Component {
+   Map<String,Integer> damageTypes;
+    public DamageResistComponent() {
+    damageTypes=new HashMap<>(6);
+    this.setAll(0);
+    }
+    public void setAll(int d){
```

Comment:
Part of the Entity-Component-System structure is that there should be no logic in the components. Instead the recommended method is to have the logic in systems instead.

So for instance, here, I would replace the whole component with simply:
```java
public Map<String, Integer> damageTypes;
```
and then move this `setAll` method into the command system.

Reply:
Sure,I will take care of that.